.github/workflows/wheels.yml

```diff
@@ -96,10 +157,57 @@ jobs:
       - uses: actions/upload-artifact@v3
         with:
           path: ./wheelhouse/*.whl
-      - name: Install twine
-        run: python -m pip install twine
-      - name: Upload to PyPI
-        run: twine upload ./wheelhouse/*.whl
-        env:
-          TWINE_PASSWORD: ${{ secrets.TWINE_PASSWORD }}
-          TWINE_USERNAME: qiskit
+      - name: Publish package distributions to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          packages-dir: wheelhouse/
+  sdist:
+    name: Build and publish terra sdist
+    runs-on: ${{ matrix.os }}
+    needs: ["build_wheels"]
```

Comment:
Can we allow the sdist to build and upload concurrently to the wheels now that the metapackage is built from this package?  We no longer need to have Terra installable to deploy the metapackage update, so maybe it's not necessary?

It's not a big deal to leave it though, I suppose.

Reply:
I left it like this to avoid a situation where a user does `pip install qiskit-terra` and has pip pull the latest version from sdist and get an error about rustc not being available. In the past with rustworkx on every release I used to get at least one person ping me asking about that error during the ~30min window between the sdist getting published and the wheel jobs for their platform getting published.

Reply:
Yeah, that's a fine enough reason to do it.